### PR TITLE
Bybit fetch deposits

### PR DESCRIPTION
**Pull request by samgermain**
Previously there was no information being returned

```
% bybit fetchDeposits USDT 1659312000000
2022-09-27T08:47:08.582Z
Node.js: v18.9.0
CCXT v1.93.87
(node:8871) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
bybit.fetchDeposits (USDT, 1659312000000)
2022-09-27T08:47:11.185Z iteration 0 passed in 421 ms

id |                                                             txid |     timestamp |                 datetime | network | address | addressTo | addressFrom | tag | tagTo | tagFrom |    type | amount | currency | status | updated | fee
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
   | a843f571651957860a15b55f87490e2f9754babb96a898504848373ed725ddd4 | 1659596295000 | 2022-08-04T06:58:15.000Z |         |         |           |             |     |       |         | deposit |   99.2 |     USDT |      3 |         |    
1 objects
2022-09-27T08:47:11.185Z iteration 1 passed in 421 ms
```

-------------

fixes: #15071

### CODE REVIEW

**Review by kroitor — DISMISSED** [review]

File: js/bybit.js [hunk 1]
```diff
@@ -4012,15 +4011,68 @@ module.exports = class bybit extends Exchange {
...
+            request['wallet_fund_type'] = 'Deposit'; // Deposit, Withdraw, RealisedPNL, Commission, Refund, Prize, ExchangeOrderWithdraw, ExchangeOrderDeposit
+            if (since <= depositMethodCutoff) {
+                method = 'privateGetV2PrivateWalletFundRecords';
+                request['start_date'] = this.yyyymmdd (since);
+            } else {
+                const startTime = since / 1000;
+                request['start_time'] = startTime;
+                const thirtyDays = 2592000;
+                if (until === undefined) {
+                    request['end_time'] = startTime + thirtyDays;
```
Comment (kroitor) [inline js/bybit.js:4027]: Can't use + for arithmetic addition here.

```suggestion
                    request['end_time'] = this.sum (startTime, thirtyDays);
```

File: js/bybit.js [hunk 2]
```diff
@@ -4012,15 +4011,68 @@ module.exports = class bybit extends Exchange {
...
+                    throw new BadRequest (this.id + ' fetchDeposits () time between since and params["until"] cannot be longer than 30 days');
+                }
+            }
+        }
+        if (until !== undefined) {
+            if (until <= depositMethodCutoff) {
+                method = 'privateGetV2PrivateWalletFundRecords';
+                request['end_date'] = this.yyyymmdd (until);
+            } else {
+                request['end_time'] = until / 1000;
```
Comment (kroitor) [inline js/bybit.js:4038]: Do they accept fractional numbers here?

Comment (samgermain) [reply js/bybit.js]: fixed in https://github.com/ccxt/ccxt/pull/15116/commits/493ea49e91b8dcbc3ea83784a9aaf7c67622e56e

**Review by sc0Vu — COMMENTED** [review]
May I ask why do we need to separate api calls before and after 2021-07-15? I didn't find in their api documentation.

**Comment by frosty00** [comment]
closing because we have moved the entire file to the v3 api

**Comment by hosseinab** [comment]
hi @frosty00
bybit version 3 ?

**Pull request closed without merging**